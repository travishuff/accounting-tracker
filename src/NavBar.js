import React from 'react'
import { Link } from '@reach/router'

import './css/nav-bar.css'
import banana from '../images/banana-logo.png'

import { Menu } from 'semantic-ui-react'

class NavBar extends React.Component {
  state = {}

  handleItemClick = (e, { name }) => this.setState({ activeItem: name })

  render() {
    const { activeItem } = this.state

    return (
      <Menu stackable>
        <Menu.Item as={Link} to="/">
          <img src={banana} className="logo" alt="Banana Tracker logo" />
          Banana Tracker
        </Menu.Item>
        <Menu.Item
          as={Link}
          to="/analytics"
          name="analytics"
          active={activeItem === 'analytics'}
          onClick={this.handleItemClick}
        >
          Analytics
        </Menu.Item>
        <Menu.Item
          as={Link}
          to="/buy"
          name="buy"
          active={activeItem === 'buy'}
          onClick={this.handleItemClick}
        >
          Buy
        </Menu.Item>
        <Menu.Item
          as={Link}
          to="/sell"
          name="sell"
          active={activeItem === 'sell'}
          onClick={this.handleItemClick}
        >
          Sell
        </Menu.Item>
      </Menu>
    )
  }
}

export default NavBar
