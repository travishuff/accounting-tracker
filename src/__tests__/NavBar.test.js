import React from 'react'
import { create } from 'react-test-renderer'

import NavBar from '../NavBar'

// Snapshot test
test('snapshot', () => {
  const c = create(<NavBar />)

  expect(c.toJSON()).toMatchSnapshot()
})
