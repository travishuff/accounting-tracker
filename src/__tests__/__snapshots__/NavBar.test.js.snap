// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="ui stackable menu"
>
  <a
    aria-current="page"
    className="item"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Banana Tracker logo"
      className="logo"
      src={Object {}}
    />
    Banana Tracker
  </a>
  <a
    className="item"
    href="/analytics"
    onClick={[Function]}
  >
    Analytics
  </a>
  <a
    className="item"
    href="/buy"
    onClick={[Function]}
  >
    Buy
  </a>
  <a
    className="item"
    href="/sell"
    onClick={[Function]}
  >
    Sell
  </a>
</div>
`;
